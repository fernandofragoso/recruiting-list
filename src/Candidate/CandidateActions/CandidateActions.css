.CandidateActions {
  display: flex;
}

.CandidateActions__button {
  text-align: center;
  border: 1px solid #CCC !important; 
  cursor:pointer;
  border:none;
  width:40px;
  height:30px;
  margin-left: 10px;
  display: flex;
  justify-content: center;
  outline: 0;
}

.CandidateActions__button:hover {
  background: #F1F2F4;
}

.CandidateActions__button--popover {
  width:200px;
  justify-content: flex-start;
}

.CandidateActions__popover-wrapper {
  position: relative;
}

.CandidateActions__popover-button {
  display: none;
  position: absolute;
  top: 30px;
  width: 200px;
  right: 10px;
  transition: display 1s;
}

.CandidateActions__popover-button--visible {
  display: block;
  transition: all 1s;
}