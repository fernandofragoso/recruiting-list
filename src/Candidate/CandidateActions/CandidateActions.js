import React, { Component } from 'react';
import './CandidateActions.css';

import staricon from '../../img/star2.svg';
import starfilledicon from '../../img/starfilled2.svg';
import dotsicon from '../../img/dots.svg';

class CandidateActions extends Component {
  
  constructor(props) {
    super(props);
    this.state = {
      showMenu: false
    }
  }
  
  render() {

    let menuclasses = "CandidateActions__popover-button";
    if (this.state.showMenu) {
      menuclasses += " CandidateActions__popover-button--visible";
    }

    let star = <img src={staricon} />
    if (this.props.favorite) {
      star = <img src={starfilledicon} />
    }

    return (
      <div className="CandidateActions">
        <button onClick={this._markAsFavorite.bind(this)} className="CandidateActions__button">
          {star}
        </button>
        <div className="CandidateActions__popover-wrapper">
          <button 
            onMouseOver={() => this._toggleOptions(!this.state.showmenu)} 
            onMouseOut={() => this._toggleOptions(false)} 
            className="CandidateActions__button">
            <img src={dotsicon} />
          </button>
          <div className={menuclasses}>
            <button 
              onClick={this._removeCandidate.bind(this)} 
              onMouseOver={() => this._toggleOptions(true)} 
              onMouseOut={() => this._toggleOptions(false)} 
              className="CandidateActions__button CandidateActions__button--popover">
              Remover
            </button>
          </div>
        </div>
      </div>
    );
  }

  _markAsFavorite() {
    this.props.onFavoriteClick();
  }

  _toggleOptions(show) {
    this.setState({
      showMenu: show
    });
  }

  _removeCandidate() {
    if (window.confirm("Deseja mesmo remover o candidato?")) {
      this.props.onRemoveClick();
    }
    this._toggleOptions(false);
  }
}

export default CandidateActions;
