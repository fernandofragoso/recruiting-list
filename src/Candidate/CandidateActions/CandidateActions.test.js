import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CandidateActions from './CandidateActions';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CandidateActions favorite='false' />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly with favorite true', () => {
  const tree = renderer.create(<CandidateActions favorite='true' />);
  expect(tree).toMatchSnapshot();
});

it('renders correctly with favorite false', () => {
  const tree = renderer.create(<CandidateActions favorite='false' />);
  expect(tree).toMatchSnapshot();
});
