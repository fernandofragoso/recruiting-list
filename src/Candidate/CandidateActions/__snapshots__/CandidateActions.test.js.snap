// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with favorite false 1`] = `
<div
  className="CandidateActions"
>
  <button
    className="CandidateActions__button"
    onClick={[Function]}
  >
    <img
      src="starfilled2.svg"
    />
  </button>
  <div
    className="CandidateActions__popover-wrapper"
  >
    <button
      className="CandidateActions__button"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        src="dots.svg"
      />
    </button>
    <div
      className="CandidateActions__popover-button"
    >
      <button
        className="CandidateActions__button CandidateActions__button--popover"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        Remover
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly with favorite true 1`] = `
<div
  className="CandidateActions"
>
  <button
    className="CandidateActions__button"
    onClick={[Function]}
  >
    <img
      src="starfilled2.svg"
    />
  </button>
  <div
    className="CandidateActions__popover-wrapper"
  >
    <button
      className="CandidateActions__button"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        src="dots.svg"
      />
    </button>
    <div
      className="CandidateActions__popover-button"
    >
      <button
        className="CandidateActions__button CandidateActions__button--popover"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        Remover
      </button>
    </div>
  </div>
</div>
`;
